lib/Dialect/FIRRTL/Import/FIRParser.cpp

```diff
@@ -4964,6 +4964,15 @@ ParseResult FIRCircuitParser::parseExtModule(CircuitOp circuit,
   if (parseParameterList(parameters) || parseRefList(portList, internalPaths))
     return failure();
 
+  if (version >= FIRVersion{4, 0, 0}) {
+    for (auto [pi, loc] : llvm::zip_equal(portList, portLocs)) {
+      if (auto ftype = dyn_cast<FIRRTLType>(pi.type)) {
```

Comment:
nit: do we need to use our type_dyn_cast here and below to be safe?

Reply:
Good catch, thanks!

Don't need to but I think I'd rather write it that way and let the constexpr thing just `dyn_cast` so no one has to ask themselves this or have subtle bugs if things change re:what is/can be an alias :+1: .

Reply:
I see, yeah I wasn't sure if that would even be legal here. Up to you if it's worth it, I just pattern match on "should this be our custom dyn_cast".